### PR TITLE
feat(sca): add text output for sca scan pre-commit

### PR DESCRIPTION
**Pull request by xblanchot-gg**
Issue: SCA-825

### CODE REVIEW

File: ggshield/sca/output/text_handler.py [hunk 1]
```diff
@@ -73,9 +98,16 @@ def process_sca_file_result(
...
+    """
+    return format_text("\nNo SCA vulnerability has been found.\n", STYLE["no_secret"])
+
+
+def no_sca_vulnerability_added() -> str:
+    """
+    Build a message if no SCA vulnerabilities were added.
     """
-    return format_text("\nNo SCA vulnerability has been found\n", STYLE["no_secret"])
+    return format_text("\nNo SCA vulnerability added.\n", STYLE["no_secret"])
```
Comment (amascia-gg) [inline ggshield/sca/output/text_handler.py:110]:
```suggestion
    return format_text("\nNo added SCA vulnerabilit.\n", STYLE["no_secret"])
```

or No SCA vulnerability has been added.

File: tests/unit/cmd/sca/test_scan.py [hunk 2]
```diff
@@ -172,3 +173,67 @@ def test_sca_scan_all_cmd(
...
+        assert result.exit_code == ExitCode.SCAN_FOUND_PROBLEMS
+        assert "> Pipfile.lock: 1 incident detected" in result.stdout
+        assert (
+            """
+Severity: Medium
+Summary: sqlparse contains a regular expression that is vulnerable to Regular Expression Denial of Service
+A fix is available at version 0.4.4
+CVE IDs: CVE-2023-30608"""
+            in result.stdout
+        )
```
Comment (amascia-gg) [inline tests/unit/cmd/sca/test_scan.py:239]: would it be possible to add some other added/removed vulns? These asserts are the same as for scan all

**Review by amascia-gg — CHANGES_REQUESTED** [review]

File: tests/unit/cmd/sca/test_scan.py (hunk 2 above)
Comment (xblanchot-gg) [reply tests/unit/cmd/sca/test_scan.py]: Added ! Good point, I realized that the text associated with removed vulns was "incident detected". Modified.

File: ggshield/sca/output/text_handler.py [hunk 3]
```diff
@@ -71,11 +108,31 @@ def process_sca_file_result(
         return result_buf.getvalue()
 
 
+def sca_file_info(
```
Comment (amascia-gg) [inline ggshield/sca/output/text_handler.py:111]: You can just use `file_info` with an optional `incident_status` argument equal to `"detected"` here I think.

File: tests/unit/cmd/sca/test_scan.py [hunk 4]
```diff
@@ -172,3 +181,135 @@ def test_sca_scan_all_cmd(
...
+        added_vulns=[
+            SCALocationVulnerability(
+                location="Pipfile.lock",
+                package_vulns=[
+                    SCAVulnerablePackageVersion(
+                        package_full_name="toto",
+                        version="1.2.3",
+                        ecosystem="pypi",
+                        vulns=[
+                            SCAVulnerability(
```
Comment (amascia-gg) [inline tests/unit/cmd/sca/test_scan.py:240]: there is going to be `ghsa_ids` here right? maybe add a `# TODO add ghsa_ids`?

**Review by amascia-gg — CHANGES_REQUESTED** [review]

File: ggshield/sca/output/text_handler.py (hunk 3 above)
Comment (xblanchot-gg) [reply ggshield/sca/output/text_handler.py:111]: Added, with an Enum in constant. I'm not fully conviced by my proposal. WDYT ?

File: ggshield/core/constants.py [hunk 5]
```diff
@@ -18,3 +19,8 @@ def _get_max_workers() -> int:
 DEFAULT_HMSL_URL = "https://api.hasmysecretleaked.com"
 AUTH_CONFIG_FILENAME = "auth_config.yaml"
 ON_PREMISE_API_URL_PATH_PREFIX = "/exposed"
+
+
+class IncidentStatus(Enum):
```
Comment (amascia-gg) [inline ggshield/core/constants.py:24]:
```suggestion
class IncidentStatus(str, Enum):
```

You need to inherit from string in order to make the string formatting work

File: ggshield/sca/output/text_handler.py (hunk 3 above)
Comment (amascia-gg) [reply ggshield/sca/output/text_handler.py]: I think it's good, what is bothering you in this?

**Review by gg-jonathangriffe — APPROVED** [review]

File: ggshield/sca/output/text_handler.py (hunk 3 above)
Comment (xblanchot-gg) [reply ggshield/sca/output/text_handler.py]: I think it's modifying some core element with technical specficities maybe. But let's do it this way !

**Review by amascia-gg — APPROVED** [review]

**Pull request merged**